packages/components/src/menu/stories/index.story.tsx

```diff
@@ -64,300 +64,337 @@ const meta: Meta< typeof Menu > = {
+						</Menu.ItemHelpText>
+					</Menu.Item>
+					<Menu.Item hideOnClick={ false }>
+						<Menu.ItemLabel>Label</Menu.ItemLabel>
+						<Menu.ItemHelpText>
+							This item doesn&apos;t close the menu on click
+						</Menu.ItemHelpText>
 					</Menu.Item>
+					<Menu.Item disabled>Disabled item</Menu.Item>
+					<Menu.Separator />
+					<Menu.Group>
+						<Menu.GroupLabel>Group label</Menu.GroupLabel>
+						<Menu.Item
+							prefix={ <Icon icon={ customLink } size={ 24 } /> }
+						>
+							<Menu.ItemLabel>With prefix</Menu.ItemLabel>
+						</Menu.Item>
+						<Menu.Item suffix="⌘S">With suffix</Menu.Item>
+						<Menu.Item
+							disabled
+							prefix={
+								<Icon icon={ formatCapitalize } size={ 24 } />
+							}
+							suffix="⌥⌘T"
+						>
+							<Menu.ItemLabel>
+								Disabled with prefix and suffix
+							</Menu.ItemLabel>
+							<Menu.ItemHelpText>And help text</Menu.ItemHelpText>
+						</Menu.Item>
+					</Menu.Group>
+				</Menu.Popover>
+			</>
+		),
+	},
+};
+
+export const WithSubmenu: Story = {
+	args: {
+		children: (
+			<>
+				<Menu.TriggerButton
+					render={
+						<Button __next40pxDefaultSize variant="secondary" />
+					}
+				>
+					Open menu
+				</Menu.TriggerButton>
+				<Menu.Popover>
+					<Menu.Item>Level 1 item</Menu.Item>
 					<Menu>
-						<Menu.SubmenuTriggerItem>
-							<Menu.ItemLabel>Submenu trigger</Menu.ItemLabel>
+						<Menu.SubmenuTriggerItem suffix="Suffix">
+							<Menu.ItemLabel>
+								Submenu trigger item with a long label
+							</Menu.ItemLabel>
 						</Menu.SubmenuTriggerItem>
 						<Menu.Popover>
 							<Menu.Item>
-								<Menu.ItemLabel>Level 3 item</Menu.ItemLabel>
+								<Menu.ItemLabel>Level 2 item</Menu.ItemLabel>
 							</Menu.Item>
 							<Menu.Item>
-								<Menu.ItemLabel>Level 3 item</Menu.ItemLabel>
+								<Menu.ItemLabel>Level 2 item</Menu.ItemLabel>
 							</Menu.Item>
+							<Menu>
+								<Menu.SubmenuTriggerItem>
+									<Menu.ItemLabel>
+										Submenu trigger
+									</Menu.ItemLabel>
+								</Menu.SubmenuTriggerItem>
+								<Menu.Popover>
+									<Menu.Item>
+										<Menu.ItemLabel>
+											Level 3 item
+										</Menu.ItemLabel>
+									</Menu.Item>
+									<Menu.Item>
+										<Menu.ItemLabel>
+											Level 3 item
+										</Menu.ItemLabel>
+									</Menu.Item>
+								</Menu.Popover>
+							</Menu>
 						</Menu.Popover>
 					</Menu>
 				</Menu.Popover>
-			</Menu>
-		</Menu.Popover>
-	</Menu>
-);
-WithSubmenu.args = {
-	...Default.args,
+			</>
+		),
+	},
 };
 
-export const WithCheckboxes: StoryFn< typeof Menu > = ( props: MenuProps ) => {
-	const [ isAChecked, setAChecked ] = useState( false );
-	const [ isBChecked, setBChecked ] = useState( true );
-	const [ multipleCheckboxesValue, setMultipleCheckboxesValue ] = useState<
-		string[]
-	>( [ 'b' ] );
+export const WithCheckboxes: Story = {
+	render: function WithCheckboxes( props: MenuProps ) {
```

Comment:
Why do we need to provide a named function here? Can't we just do:

```
	render: ( props: MenuProps ) => (
		<>
			//...
		</>
	),
```

Reply:
I used named functions to prevent warnings about using hooks outside of a component.

Reply:
I see. The rules of hooks are broken because React doesn't know how to statically treat hooks inside a dynamically assigned function component like this - it just can't possibly guarantee if `render()` is not going to be passed around, which would break rules of hooks. 

To me that's just another reason not to use this `render()` syntax and to use `args.children` if possible. 

Alternatively, we could move the component declaration outside of `render()` and just pass an instance here - would that work better? The hack with tricking the linter bit us back and we had to change it, which is why I'm hesitant to do it once more. 

I'm leaving this to your judgment, as I don't have particularly strong feelings.

Reply:
My understanding was that the hooks linter simply can't know that `render: () => {}` is meant to be a React component. `render: function UppercaseName() {}` just tells the linter that it's meant to be a component, like any other function with a uppercased name.

What we're doing for `render` isn't that different from the way we can define subcomponents, e.g.:

```jsx
// this is fine
Menu.Submenu = () => { /* something with hooks */ };

// this is not fine because lowercase
Menu.submenu = () => { /* something with hooks */ };

// would've been fine if the `render` field had been uppercase like this 😅
Default.Render = () => { /* something with hooks */ };
```

So while I can't say that this isn't a "hack", I don't think it's hacky in a way that's unsafe?

Reply:
I've elaborated on it [here](https://github.com/WordPress/gutenberg/pull/68204#issuecomment-2559706018), and it's definitely not unsafe, and we can move forward with it for Storybook.